shell_integration/MacOSX/OwnCloudFinder/RequestManager.m

```diff
@@ -16,26 +16,32 @@
 #import "IconCache.h"
 #import "RequestManager.h"
 
-#define READ_TAG 2422
+static OwnCloudFinderRequestManager* sharedInstance = nil;
 
-static RequestManager* sharedInstance = nil;
-
-@implementation RequestManager
+@implementation OwnCloudFinderRequestManager
 
 - (id)init
 {
 	if ((self = [super init]))
 	{
-		_socket = [[GCDAsyncSocket alloc] initWithDelegate:self delegateQueue:dispatch_get_main_queue()];
+		// For the sake of allowing both the legacy and the FinderSync extensions to work with the same
+		// client build, use the same server name including the Team ID even though we won't be sandboxed.
+		NSBundle *extBundle = [NSBundle bundleForClass:[self class]];
+		// This was added to the bundle's Info.plist to get it from the build system
+		NSString *socketApiPrefix = [extBundle objectForInfoDictionaryKey:@"SocketApiPrefix"];
+		NSString *serverName = [socketApiPrefix stringByAppendingString:@".socketApi"];
+		// NSLog(@"OwnCloudFinderRequestManager serverName %@", serverName);
 
-		_isConnected = NO;
+		_syncClientProxy = [[SyncClientProxy alloc] initWithDelegate:self serverName:serverName];
 
 		_registeredPathes = [[NSMutableDictionary alloc] init];
 		_requestedPaths = [[NSMutableSet alloc] init];
 
 		_shareMenuTitle = nil;
 
-		[NSTimer scheduledTimerWithTimeInterval:5 target:self selector:@selector(start) userInfo:nil repeats:YES];
+		// The NSConnection will block until the distant object came back and this creates a loop hanging Finder.
+		// Start from a timer to have time to unwind the stack first.
+		[NSTimer scheduledTimerWithTimeInterval:0 target:_syncClientProxy selector:@selector(start) userInfo:nil repeats:NO];
```

Comment:
Btw, could also use ```performSelector``` with ```afterDelay``` (no need to change though)

http://woboq.com/blog/objective-c-for-qt-developers.html